js/index.js

```diff
@@ -17,18 +17,9 @@ function showPreview() {
 	if (!tootId) {
 		return;
 	}
-	const tootUrl = `${instanceFull}/api/v1/statuses/${tootId}`;
-	fetch(tootUrl)
-		.then(async response => {
-			if (response.ok) {
-				const toot = await response.json();
-				const tootDiv = impl.createTootDiv(toot);
-				$("card-preview").innerHTML = tootDiv.outerHTML;
-			} else {
-				throw new Error(`Request failed: ${response.status}`);
-			}
-		})
-		.catch(err => console.error(err));
+	const toot = await impl.fetchJsonAndCheck(`${instanceFull}/api/v1/statuses/${tootId}`);
```

Comment:
共通化してしまったため、 fetch に失敗した場合に null が返ってきます
null チェックが必要です
```suggestion
	const toot = await impl.fetchJsonAndCheck(`${instanceFull}/api/v1/statuses/${tootId}`);
	if (!toot) {
		return;
	}
```